packages/jupyter-ai/src/chat_handler.ts

```diff
@@ -164,6 +140,41 @@ export class ChatHandler implements IDisposable {
     (value: AiService.AgentChatMessage) => void
   > = {};
 
+  private _onClose(reject: any) {
+    reject(new Error("Chat UI websocket disconnected"))
+    console.error("Chat UI websocket disconnected")
+    setTimeout(async () => await this._initialize(), 500);
+  }
+
+  private _initialize(): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      if (this.isDisposed) {
+        return;
+      }
+      console.log("Creating a new websocket connection...");
```

Comment:
```suggestion
      console.log("Creating a new websocket connection for chat ...");
```

We should clarify in the console which websocket connection we're creating, if this log statement is necessary at all